### PR TITLE
EPMLSTRCMW-292 Refactor custom exceptions for RunService

### PR DESCRIPTION
**Pull request by DruzhininPavel**

### CODE REVIEW

File: services/src/main/scala/cromwell/pipeline/service/RunService.scala [hunk 1]
```diff
@@ -38,36 +48,69 @@ object RunService {
...
+                  cmwlWorkflowId = request.cmwlWorkflowId
+                )
               )
-            )
-          case None => Future.failed(new RuntimeException("run with this id doesn't exist"))
+              .recoverWith { case _ => internalError("update run") }
+          case None => Future.failed(NotFound())
+        }
+
+      private def internalError(action: String): Future[Nothing] =
```
Comment (Nikolai-Isaev) [inline services/src/main/scala/cromwell/pipeline/service/RunService.scala:105]: That method name does not look like an action. Maybe "getInternalError" or smth like that would be better

**Review by Nikolai-Isaev — APPROVED** [review]
Nice

File: services/src/main/scala/cromwell/pipeline/service/RunService.scala [hunk 2]
```diff
@@ -38,36 +48,69 @@ object RunService {
           userId = userId,
           results = runCreateRequest.results
         )
-        projectService.getUserProjectById(projectId, userId).flatMap(_ => runRepository.addRun(newRun))
+        projectService
+          .getUserProjectById(projectId, userId)
+          .recoverWith {
+            case e: ProjectServiceException => serviceErrorMapper(e)
+          }
+          .flatMap(_ => runRepository.addRun(newRun))
```
Comment (Nikolai-Isaev) [inline services/src/main/scala/cromwell/pipeline/service/RunService.scala:56]: Can `transform` be used here?

File: services/src/main/scala/cromwell/pipeline/service/RunService.scala (hunk 1 above)
Comment (DruzhininPavel) [reply services/src/main/scala/cromwell/pipeline/service/RunService.scala:105]: This name reflects the results you get and is familiar. We do not pass the result outside the class boundaries. In addition, in other classes, it is also named the same, I think it is not worth changing traditions :)

File: services/src/main/scala/cromwell/pipeline/service/RunService.scala (hunk 2 above)
Comment (DruzhininPavel) [reply services/src/main/scala/cromwell/pipeline/service/RunService.scala:56]: Do you have a specific example ?

**Review by gominasian — APPROVED** [review]
Looks good to me

**Review by myazinn — APPROVED** [review]

**Review by GrigoriiBerezin — APPROVED** [review]
Nice

**Pull request merged**